src/network/networkd-link.c

```diff
@@ -1567,6 +1571,13 @@ static int link_up(Link *link) {
         if (r < 0)
                 return log_link_error_errno(link, r, "Could not allocate RTM_SETLINK message: %m");
 
+        /* set it free if it's not enslaved with networkd; bond slave gains IFF_UP once master is set */
```

Comment:
But if we set the bond slave down manually and then restart networkd, it will unset the master if we don't check `link->network->bond` here.